### DISCUSSION TITLE
Default --app-dir makes __file__ ugly

### DISCUSSION THREAD

**Discussion by mila** [Kludex/uvicorn#1814] [Potential Issue] [answered]
I use a slightly modified quickstart example to output value of the `__file__` variable. 

```python
async def app(scope, receive, send):
    print(__file__)
    assert scope['type'] == 'http'
    await send({
        'type': 'http.response.start',
        'status': 200,
        'headers': [
            [b'content-type', b'text/plain'],
        ],
    })
    await send({
        'type': 'http.response.body',
        'body': __file__.encode(),
    })
```

When I run the app using `uvicorn example:app`, I get a path like `/Users/mpojman/Projects/uvicorn-sandbox/./example.py`. Note the unnecessary dot. This is caused by `--app-dir`, which defaults to `"."`.

I think that an empty string could be a better default value for this option. When I run the app using `uvicorn example:app --app-dir ""`, `__file__` is `/Users/mpojman/Projects/uvicorn-sandbox/example.py` as it should be.

The empty string would be consistent with Python behavior:
```
$ python -c 'import sys; print(repr(sys.path[0]))'
''
```

**Comment by Kludex** [comment, marked as answer]
If the behavior is the same, and you can point me out the discussion we had when it was introduced on a PR, I can review it.

**Reply by mila** [reply]
Thank you @Kludex. 

- Pull requests is here: https://github.com/encode/uvicorn/pull/1835
- `--app-dir` was added in https://github.com/encode/uvicorn/pull/619. It seems to me that the dot was a natural choice for the current directory, so the empty string did not even come to mind.
